storm-core/test/jvm/org/apache/storm/utils/UtilsTest.java

```
@@ -0,0 +1,213 @@
+    }
+
+    private Map emptyMockMap() {
+        return new HashMap<String, Object>();
+    }
+
+    @Test
+    public void isZkAuthenticationConfiguredTopologyTest() {
+        Assert.assertFalse(
+            "Returns null if given null config", 
+            Utils.isZkAuthenticationConfiguredTopology(null));
+
+        Assert.assertFalse(
+            "Returns false if scheme key is missing", 
+            Utils.isZkAuthenticationConfiguredTopology(emptyMockMap()));
+
+        Assert.assertFalse(
+            "Returns false if scheme value is null", 
+            Utils.isZkAuthenticationConfiguredTopology(topologyMockMap(null)));
+
+        Assert.assertTrue(
+            "Returns true if scheme value is string", 
+            Utils.isZkAuthenticationConfiguredTopology(topologyMockMap("foobar")));
+    }
+
+    @Test
+    public void isZkAuthenticationConfiguredStormServerTest() {
+        Assert.assertFalse(
+            "Returns false if given null config",
+            Utils.isZkAuthenticationConfiguredStormServer(null));
+
+        Assert.assertFalse(
+            "Returns false if scheme key is missing",
+            Utils.isZkAuthenticationConfiguredStormServer(emptyMockMap()));
+
+        Assert.assertFalse(
+            "Returns false if scheme value is null",
+            Utils.isZkAuthenticationConfiguredStormServer(serverMockMap(null)));
+
+        Assert.assertTrue(
+            "Returns true if scheme value is string",
+            Utils.isZkAuthenticationConfiguredStormServer(serverMockMap("foobar")));
+    }
+
+    @Test
+    public void isZkAuthenticationConfiguredStormServerWithPropertyTest() {
+        String key = "java.security.auth.login.config";
+        String oldValue = System.getProperty(key);
+        try {
+            System.setProperty("java.security.auth.login.config", "anything");
+            Assert.assertTrue(Utils.isZkAuthenticationConfiguredStormServer(emptyMockMap()));
+        } catch (Exception ignore) {
```

We don't want to ignore this exception.  Let it fail the test if need be.